File: mobile_thru_project_server_full/mobile_thru_project_server_full/src/main/java/com/ssafy/greenplate/model/service/CouponService.java
package com.ssafy.greenplate.model.service;

import java.util.List;

import com.ssafy.greenplate.model.dto.Coupon;

public interface CouponService {
    /**
     * Coupon을 등록한다.
     * 
     * @param coupon
     */
    void addCoupon(Coupon coupon);

    /**
     * id에 해당하는 Coupon을 반환한다.
     * 
     * @param id
     * @return
     */
    Coupon selectCoupon(Integer id);

    /**
     * id에 해당하는 Coupon을 삭제한다.
     * 
     * @param id
     */
    void removeCoupon(Integer id);

    /**
     * Coupon 목록읅 반환한다.
     * 
     * @param id
     * @return
     */
    List<Coupon> selectAllCoupons(String userId);
}

File: green_plate_project/app/src/main/java/com/ssafy/green_plate/api/CouponService.kt
package com.ssafy.green_plate.api

import com.ssafy.green_plate.dto.Coupon
import com.ssafy.green_plate.dto.Order
import retrofit2.http.*

interface CouponService {

    @POST("rest/coupon")
    suspend fun addCoupon(@Body body: Coupon)
    @GET("rest/coupon/{userId}")
    suspend fun getCoupon(@Path("userId") userId: String) : List<Coupon>
    @DELETE("rest/coupon/{id}")
    suspend fun deleteCoupon(@Path("id") id: String)
}